### PR TITLE
[0002] Root signatures: Fix example and reformat metadata schema

### PR DESCRIPTION
**Pull request by damyanp**
Move the example metadata to the main proposal section.

Update the layout of the schema so that it is ordered more from top-to-bottom and reformat to be easier to read.

Also fixed the metadata example to match the schema.

### CODE REVIEW

File: proposals/0002-root-signature-in-clang.md [hunk 1]
```diff
@@ -471,116 +489,173 @@ The additional semantic rules not already covered by the grammar are listed here
...
-!8 = !{ !"StaticSampler", i32 1, i32 0, ... } ; register 1, space 0, (additional params omitted)
 ```
 
+The function/root signature associates a function (the first operand) with a
+reference to a root signature (the second operand).
+
+#### Root Signature
+
+```LLVM
+!3 = !{!4, !5, !6, !7 }
```
Comment (inbelic) [inline proposals/0002-root-signature-in-clang.md:533]:
```suggestion
!3 = !{ !4, !5, !6, !7 }
```

File: proposals/0002-root-signature-in-clang.md [hunk 2]
```diff
@@ -393,6 +393,24 @@ stored as metadata nodes, identified by named metadata. The metadata format
...
 
+While the attribute is attached to a function, the metadata collects all the
+root signatures together, with the initial metadata associating the root
+signatures with functions.
+
+Example for same root signature as above:
+
+```llvm
+!dx.rootsignatures = !{!2} ; list of function/root signature pairs
+!2 = !{ptr @main, !3 } ; function, root signature
```
Comment (inbelic) [inline proposals/0002-root-signature-in-clang.md:404]:
```suggestion
!2 = !{ ptr @main, !3 } ; function, root signature
```
Not sure if needed, but everywhere else seems to have a space. There is another instance if this is the case on line 524

File: proposals/0002-root-signature-in-clang.md [hunk 3]
```diff
@@ -471,116 +489,173 @@ The additional semantic rules not already covered by the grammar are listed here
...
+
+```LLVM
+!4 = { !"RootFlags", i32 1 }
+```
+
+Operands:
+
+* i32: the root signature flags ([D3D12_ROOT_SIGNATURE_FLAGS](d3d12_root_signature_flags))
+
+
```
Comment (inbelic) [inline proposals/0002-root-signature-in-clang.md:567]: Extra line

File: proposals/0002-root-signature-in-clang.md [hunk 4]
```diff
@@ -471,116 +489,173 @@ The additional semantic rules not already covered by the grammar are listed here
...
+#### Root Flags
+
+```LLVM
+!4 = { !"RootFlags", i32 1 }
+```
+
+Operands:
+
+* i32: the root signature flags ([D3D12_ROOT_SIGNATURE_FLAGS](d3d12_root_signature_flags))
```
Comment (inbelic) [inline proposals/0002-root-signature-in-clang.md:565]:
```suggestion
* i32: the root signature flags ([D3D12_ROOT_SIGNATURE_FLAGS][d3d12_root_signature_flags])
```
Discrepancy with the other flag links

File: proposals/0002-root-signature-in-clang.md [hunk 5]
```diff
@@ -471,116 +489,173 @@ The additional semantic rules not already covered by the grammar are listed here
...
+!9 = !{ !"UAV", i32 5, i32 1, i32 10, i32 0 }
+```
+
+Operands:
+
+* string: type of range - "SRV", "UAV", "CBV" or "Sampler"
+* i32: number of descriptors in the range
+* i32: base shader register
+* i32: register space
+* i32: descriptor range flgs ([D3D12_DESCRIPTOR_RANGE_FLAGS][d3d12_descriptor_range_flags])
```
Comment (inbelic) [inline proposals/0002-root-signature-in-clang.md:625]:
```suggestion
* i32: descriptor range flags ([D3D12_DESCRIPTOR_RANGE_FLAGS][d3d12_descriptor_range_flags])
```

File: proposals/0002-root-signature-in-clang.md [hunk 6]
```diff
@@ -471,116 +489,173 @@ The additional semantic rules not already covered by the grammar are listed here
...
+
+#### Static Samplers
+
+```LLVM
+!6 = !{ !"StaticSampler", i32 1, i32 0, ... }; remaining operands omitted for space
+```
+
+Operands:
+* i32: Filter ([D3D12_FILTER](d3d12_filter))
```
Comment (inbelic) [inline proposals/0002-root-signature-in-clang.md:634]: These also use regular braces as opposed to square. Not sure which one is correct but I'll note it to double-check.

**Review by inbelic — APPROVED** [review]
I was able to follow and understand the descriptions well.

**Review by coopp — APPROVED** [review]
Looks good to me.,  I like the additional cleanups here.

**Pull request merged**